textpattern/include/txp_prefs.php

```diff
@@ -120,6 +120,10 @@ function prefs_save()
         if (!isset($post[$name]) || !has_privs('prefs.'.$event)) {
             continue;
         }
+        
+        if (is_array($post[$name])) {
+            $post[$name] = implode(',', $post[$name]);
```

Comment:
Safe - I think yes, because its items are only compared against `$array` keys, which are under control. A little restrictive, sure, but many txp tokens are comma-separated lists. JSON would be a safer option.